### PR TITLE
Interface

### PR DESCRIPTION
**Pull request by ArtPoon**

### CODE REVIEW

**Comment by ArtPoon** [comment]
Did local merge of `dev` into `interface`, but am running into this error:
```python
Starting 3Seq Analysis
Finished 3Seq Analysis
Starting GENECONV Analysis
Finished GENECONV Analysis
Setting up bootscan analysis...
Traceback (most recent call last):
  File "/home/art/git/OpenRDP/venv/bin/openrdp", line 4, in <module>
    __import__('pkg_resources').run_script('OpenRDP==0.0.1', 'openrdp')
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 672, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1472, in run_script
    exec(code, namespace, namespace)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/EGG-INFO/scripts/openrdp", line 35, in <module>
    results = openrdp.openrdp(args.infile, args.outfile, cfg=args.cfg,
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/__init__.py", line 300, in openrdp
    results = scanner.run_scans(aln)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/__init__.py", line 181, in run_scans
    tmethods.append(a['method'](alignment, settings=settings, quiet=self.quiet))
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/bootscan.py", line 18, in __init__
    self.set_options_from_config(settings)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/bootscan.py", line 59, in set_options_from_config
    self.np = int(settings['np'])
KeyError: 'np'
```

**Comment by ArtPoon** [comment]
The configuration file `default.ini` has an `np` value defined:
```
[Bootscan]
max_pvalue = 0.05
win_size = 200
step_size = 20
num_replicates = 100
random_seed = 3
cutoff_percentage = 0.7
; scan using distances, UPGMA trees, neighbour joining trees
scan = distances
; p_value can be calculated by using binomial, chi-squared, or bootstrap value
p_value_calculation = binomial
; model can be Jukes-Cantor, Kimura
model = Jukes-Cantor
np = 2
```

however when I run `configparser` and dump the contents as a dict, I get the following:
```python
{'General': {'circular_genome': False, 'comparison_correction': 'bonferroni'}, 'permutation_options': {'num_permutations': 0}, 'data_processing': {'min_num_detecting_events': 1}, 'RDP': {'max_pvalue': '0.05', 'reference_sequence': 'None', 'window_size': 30, 'min_identity': 0, 'max_identity': 100}, 'Geneconv': {'indels_as_polymorphisms': True, 'mismatch_penalty': 1, 'min_len': 1, 'min_poly': 2, 'min_score': 2, 'max_num': 1}, 'Bootscan': {'max_pvalue': '0.05', 'win_size': 200, 'step_size': 20, 'num_replicates': 100, 'random_seed': 3, 'cutoff_percentage': '0.7', 'scan': 'distances', 'p_value_calculation': 'binomial', 'model': 'Jukes-Cantor', 'trans_transv': 2, 'coeff_variation': 1, 'freq_from_aln': True, 'a_freq': '0.25', 'g_freq': '0.25', 'c_freq': '0.25', 't_freq': '0.25'}, 'MaxChi': {'max_pvalue': '0.05', 'win_size': 200, 'strip_gaps': False, 'fixed_win_size': True, 'num_var_sites': 70, 'frac_var_sites': '0.1'}, 'Chimaera': {'max_pvalue': '0.05', 'win_size': 200, 'strip_gaps': False, 'fixed_win_size': True, 'num_var_sites': 60, 'frac_var_sites': '0.1'}, 'Siscan': {'max_pvalue': '0.05', 'win_size': 200, 'step_size': 20, 'strip_gaps': True, 'pvalue_perm_num': 1100, 'scan_perm_num': 100, 'random_seed': 3, 'fast_scan': True, 'fourth_seq_sel': 'outlier'}}
```

**Comment by ArtPoon** [comment]
Ok the problem was that I was forgetting to use a `-c` flag to specify the non-default config file
Different problem now:
```
Scanning triplet 1 / 4
Traceback (most recent call last):
  File "/home/art/git/OpenRDP/venv/bin/openrdp", line 4, in <module>
    __import__('pkg_resources').run_script('OpenRDP==0.0.1', 'openrdp')
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 672, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1472, in run_script
    exec(code, namespace, namespace)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/EGG-INFO/scripts/openrdp", line 35, in <module>
    results = openrdp.openrdp(args.infile, args.outfile, cfg=args.cfg,
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/__init__.py", line 301, in openrdp
    results = scanner.run_scans(aln)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/__init__.py", line 194, in run_scans
    tmethod.execute(triplet)
  File "/home/art/git/OpenRDP/venv/lib/python3.9/site-packages/OpenRDP-0.0.1-py3.9.egg/openrdp/bootscan.py", line 165, in execute
    (i, trp) = arg
TypeError: cannot unpack non-iterable Triplet object
```

File: openrdp/__init__.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+        trp_count = 1
+        for trp in generate_triplets(alignment):
+            triplet = Triplet(alignment, self.seq_names, trp)
+            self.print("Scanning triplet {} / {}".format(trp_count, total_num_trps))
+            trp_count += 1
+            for alias, tmethod in tmethods.items():
+                if alias == 'bootscan':
+                    continue
+                tmethod.execute(triplet)
```
Comment (GopiGugan) [inline openrdp/__init__.py:199]: If the `bootscan` method is being used, the `execute_all()` function generates the triplets, so there wouldn't be a need to iterate over the triplets again

Comment (ArtPoon) [reply openrdp/__init__.py:199]: got it, pulled in your code from `scripts/run_scans.py`

**Pull request merged**